### DISCUSSION TITLE
UWP Access to SQLite Database on a Network Drive

### DISCUSSION THREAD

**Discussion by KevinConManApp** [xamarin/Xamarin.Forms#14732] [Q&A]
Hi There,
I have developed a Xamarin Forms App that accesses a SQLite Database via sqlite-net-pcl. For the Android and iOS apps, I copy a centralised database file to the local device and sync it back to the central one as required.
However, for the UWP app I am connecting directly to the centralised one. It sits in Azure File Storage that is mapped as a network drive - W: on the Win10 Device that the UWP app runs on.
While this connects fine when running the app in VS2019, when I deploy a package to another win10 device, I continually get a connection error that appears to be related to access permissions.
I have followed advice regarding 'rescap: broadFileSystemAccesss' and can see the database as a Windows.Storage file but suspect it may relate to being denied System.IO access. But do not understand why I can connect in Debug mode, which also works when I debug to a remote Win10 device.
Any ideas would be most welcome.
Regards,
Kevin.

**Comment by MitchBomcanhao** [comment]
maybe related, maybe not - I can run the DEBUG variant of our UWP app when connected to the company VPN, but the same app in release mode does not work (note that other UWP apps also cannot connect to the internet - the microsoft store app is a good example of that).

**Reply by KevinConManApp** [reply]
Thanks for the response, Mitch.
Your experience with DEBUG on a UWP app would appear to be related. I managed to work around the issue by assigning a rescap:Capability of 'developmentModeNetwork' after which opening the SQLite Database on a Network drive worked fine. It means the app needs Developer Mode turned on but this is OK for the desktop scenarios we are deploying to.
Regards,
Kevin.
